src/ErrorBoundary.ts

```diff
@@ -46,8 +54,14 @@ export class ErrorBoundary extends Component<
     }
   }
 
+  componentWillUnmount() {
+    window.removeEventListener("error", handleSuppressLogging);
+  }
+
   componentDidCatch(error: Error, info: ErrorInfo) {
     this.props.onError?.(error, info);
+    window.addEventListener("error", () => {}); // Some test cases will fail without this line, somehow.
```

Comment:
We can't do this. It would break uncaught exception behavior (see https://github.com/facebook/react/pull/28515) and cause a memory leak (since `window` would have a reference to this error boundary instance and anything that it references as well).

Reply:
Moving forward, I guess I would dive deeper into the React Internals and try to understand how the error handling works in there. Thanks.